File: sources/com/s8/web/front/carbide/navbar/NavbarMenu.java
package com.s8.web.front.carbide.navbar;

import com.s8.io.bohr.neon.core.NeBranch;
import com.s8.io.bohr.neon.core.NeObject;
import com.s8.io.bohr.neon.functions.none.VoidNeFunction;
import com.s8.io.bohr.neon.lambdas.none.VoidLambda;
import com.s8.web.front.carbide.icons.S8FlatIcon;


/**
 * 
 * @author pierreconvert
 *
 */
public class NavbarMenu extends NeObject {

	
	/**
	 * 
	 * @param branch
	 */
	public NavbarMenu(NeBranch branch) {
		super(branch, "/s8-web-front/carbide/navbar/NavbarMenu");
	}
	
	
	public NavbarMenu(NeBranch branch, S8FlatIcon icon, String name) {
		super(branch, "/s8-web-front/carbide/navbar/NavbarMenu");
		setIcon(icon);
		setName(name);
		onSelected(() -> {
			System.out.println("I'm selected");
		});
	}
	
	
	/**
	 * 
	 * @param menus
	 */
	public void setIcon(S8FlatIcon icon) {
		vertex.setUInt16Field("icon", icon.code);
	}
	
	
	/**
	 * 
	 * @param name
	 */
	public void setName(String name) {
		vertex.setStringUTF8Field("name", name);
	}
	
	
	
	public void onSelected(VoidLambda lambda) {
		vertex.setVoidMethodLambda("on-selected", lambda);
	}
	
	
	/**
	 * 
	 * @param function
	 */
	public void onSelectedMethod(VoidNeFunction function) {
		vertex.setVoidMethod("on-selected", function);
	}
	
}

File: web-sources/s8-web-front/carbide/navbar/NavbarMenu.js

import { S8WebFront } from "/s8-web-front/S8WebFront.js";
import { NeObject } from "/s8-io-bohr-neon/NeObject.js";


/**
 * 
 */
S8WebFront.CSS_import('/s8-web-front/carbide/navbar/Navbar.css');


/**
 * 
 */
export class NavbarMenu extends NeObject {

    

    /**
     * @type{number}
     */
    index;


    constructor(){
        super();
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("navbar-menu");


        this.labelNode = document.createElement("div");
        this.labelNode.classList.add("navbar-menu-label");
        this.wrapperNode.appendChild(this.labelNode);

        this.iconNode = document.createElement("div");
        this.iconNode.classList.add("navbar-menu-icon");
        this.labelNode.appendChild(this.iconNode);

        this.nameNode = document.createElement("div");
        this.nameNode.classList.add("navbar-menu-name");
        this.labelNode.appendChild(this.nameNode);
        
    }




    changeSelection(isSelected){
        if(isSelected){
            this.wrapperNode.setAttribute("selected", '');
        }
        else{
            this.wrapperNode.removeAttribute("selected");
        }      
    }

    
    /**
     * 
     * @returns 
     */
    getEnvelope(){
        return this.wrapperNode;
    }


    /**
     * icon
     * @param {*} code 
     */
    S8_set_icon(code){
        S8WebFront.SVG_insertByCode(this.iconNode, code, 16, 16);
    }


    /**
     * name
     * @param {*} name 
     */
    S8_set_name(name){
       this.nameNode.innerHTML = name;
    }

    
    /**
     * 
     */
    S8_render(){
    }


    /**
     * 
     */
    S8_dispose(){
    }


}